src/check.ts

```diff
@@ -7,6 +7,26 @@ import {plural} from './render';
 
 const USER_AGENT = `${pkg.name}/${pkg.version} (${pkg.bugs.url})`;
 
+interface AnnotationContents {
+    reason?: string,
+    message: {
+        code: string,
+        level: string,
+        message: string,
+        rendered: string,
+        spans: MessageSpan[],
+    },
+}
+
+interface MessageSpan {
```

Comment:
Do you mind to rename it into the `DiagnosticSpan` to match the rust version: https://github.com/rust-lang/rust/blob/b7820b28937ba875521a2525cf75223c80faf9f2/src/libsyntax/json.rs#L141 ?